### PR TITLE
fix(DTFS2-6046): BSS dates validation

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue:
* Cover Start Date month did not show correct error
* Conversion date did not show error if invalid dates

# Changes made:
* Added validation for conversion date
* Added some extra tests

### CODE REVIEW

File: portal-api/api-tests/v1/validation/conversion-rate-date.api-test.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+describe('validation - conversion rate date', () => {
+  const deal = {
+    details: {
+      submissionDate: null,
+    }
+  };
+
+  const errorList = {};
+
+  it('should return validation error if the conversion date year has a symbol in it', () => {
```
Comment (abhi-markan) [inline portal-api/api-tests/v1/validation/conversion-rate-date.api-test.js:12]: Can we please add a test when month is `00` and year is `0000` please.

File: portal-api/src/v1/validation/bond-rules/facility-stage-issued-rules/requested-cover-start-date.js [hunk 2]
```diff
@@ -60,43 +60,44 @@ module.exports = (submittedValues, deal, errorList) => {
...
-      if (coverDayValidation.error && requestedCoverStartDateDay) {
-        // error object does not exist if no errors in validation
-        newErrorList.requestedCoverStartDate = {
-          text: 'The day for the requested Cover Start Date must include 1 or 2 numbers',
-          order: orderNumber(newErrorList),
-        };
-      }
+    }
+    if (coverDayValidation.error && requestedCoverStartDateDay) {
+      // error object does not exist if no errors in validation
```
Comment (abhi-markan) [inline portal-api/src/v1/validation/bond-rules/facility-stage-issued-rules/requested-cover-start-date.js:65]: I am not entirely sure what does this comment mean please?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, few minor comments to address please.

File: portal-api/api-tests/v1/validation/conversion-rate-date.api-test.js (hunk 1 above)
Comment (Zainzzkk) [reply portal-api/api-tests/v1/validation/conversion-rate-date.api-test.js:12]: done

File: portal-api/src/v1/validation/bond-rules/facility-stage-issued-rules/requested-cover-start-date.js (hunk 2 above)
Comment (Zainzzkk) [reply portal-api/src/v1/validation/bond-rules/facility-stage-issued-rules/requested-cover-start-date.js]: done

**Review by abhi-markan — APPROVED** [review]
LGTM

**Pull request merged**